google/cloud/bigquery/table.py

```diff
@@ -1635,7 +1647,10 @@ def _is_almost_completely_cached(self):
         This is useful to know, because we can avoid alternative download
         mechanisms.
         """
-        if self._first_page_response is None:
+        if (
+            not hasattr(self, "_first_page_response")
```

Comment:
Because we set at line 1591 `self._first_page_response = first_page_response`, this attribute will always exist? Maybe we can check whether the value is None or not.

Reply:
This also was needed for some tests where we have a mock row iterator but want to test with a real implementation of this method.